apps/src/studio/blocks.js

```diff
@@ -1751,10 +1752,23 @@ exports.install = function(blockly, blockInstallOptions) {
   };
 
   generator.functional_start_setFuncs = function() {
-    // For the current design, this doesn't need to generate any code.
-    // Though we pass in a function, we're not actually using that passed in
-    // function, and instead depend on a function of the required name existing
-    // in the global space. This may change in the future.
+    // For each of our inputs (i.e. update-target, update-danger, etc.) get
+    // the attached block and figure out what it's function name is. Store
+    // that on bigGameInfo so we can know what functions to call later.
+    this.inputList.forEach(function (input) {
+      if (input.type !== Blockly.FUNCTIONAL_INPUT) {
+        return;
+      }
+      var inputBlock = this.getInputTargetBlock(input.name);
+      if (!inputBlock) {
+        return;
+      }
+      var inputBlockName = inputBlock.getTitleValue('NAME');
+      var functionName = Blockly.JavaScript.variableDB_.getName(inputBlockName,
```

Comment:
private accessor alert (variableDB_). It looks like this isn't the only place we do this in Blockly. This would seem to indicate that we should probably create a public getter at some point, but I don't have the desire right now and I'm not making the code any worse by introducing this =P